activerecord/lib/active_record/connection_adapters/mysql/column.rb

```diff
@@ -5,7 +5,7 @@ class Column < ConnectionAdapters::Column # :nodoc:
         delegate :extra, to: :sql_type_metadata, allow_nil: true
 
         def unsigned?
-          /\bunsigned\z/.match?(sql_type)
+          !/\A(?:enum|set)\b/.match?(sql_type) && /\bunsigned\b/.match?(sql_type)
```

Comment:
I don't get why this is needed. Does `enum` and `set` types allow to be defined as `unsigned`?